src/renderer/base/CSSLengthPercentage.cpp

```diff
@@ -0,0 +1,78 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+#include "precomp.h"
+
+#include "../inc/CSSLengthPercentage.h"
+
+CSSLengthPercentage CSSLengthPercentage::FromString(const wchar_t* str)
+{
+    auto& errnoRef = errno; // Nonzero cost, pay it once.
+    errnoRef = 0;
+
+    wchar_t* end;
+    auto value = std::wcstof(str, &end);
+
+    if (str == end || errnoRef == ERANGE)
+    {
+        return {};
+    }
+
+    auto referenceFrame = ReferenceFrame::FontSize;
+
+    if (*end)
+    {
+        if (wcscmp(end, L"%") == 0)
+        {
+            value /= 100.0f;
+        }
+        else if (wcscmp(end, L"px") == 0)
+        {
+            referenceFrame = ReferenceFrame::Absolute;
+            value /= 96.0f;
+        }
+        else if (wcscmp(end, L"pt") == 0)
+        {
+            referenceFrame = ReferenceFrame::Absolute;
+            value /= 72.0f;
+        }
+        else if (wcscmp(end, L"ch") == 0)
+        {
+            referenceFrame = ReferenceFrame::AdvanceWidth;
+        }
+        else
+        {
+            return {};

```

Comment:
silent discard - OK?

i notice we have no way of expressing a failure to parse this via the settings warning dialog

Reply:
It's not okay, but I tried the alternative by implementing a WinRT runtimeclass for `CSSLengthPercentage` and using that as the value for CellWidth/Height and that was quite the mess. I would not mind adding an extra check during the settings loading. I could call `CSSLengthPercentage::FromString` inside the `FontConfig::LayerJson` to check if it's valid since `wcstof` and `wcscmp` aren't /that/ expensive. But I wasn't personally convinced it's worth it (by asking myself "Would anyone notice if I don't check this?").